diff --git a/ReactKonvaCore.d.ts b/ReactKonvaCore.d.ts
index 1c3fa21193c3f9afc266597f4ad047258d8c3b87..abe4564693fa546a7d4541dde7785c70b5079dbc 100644
--- a/ReactKonvaCore.d.ts
+++ b/ReactKonvaCore.d.ts
@@ -77,28 +77,28 @@ export var Label: KonvaNodeComponent<Konva.Label, Konva.LabelConfig>;
 
 /** Shapes */
 export var Rect: KonvaNodeComponent<Konva.Rect, Konva.RectConfig>;
-export var Circle: KonvaNodeComponent<Konva.Circle, Konva.CircleConfig>;
-export var Ellipse: KonvaNodeComponent<Konva.Ellipse, Konva.EllipseConfig>;
-export var Wedge: KonvaNodeComponent<Konva.Wedge, Konva.WedgeConfig>;
-export var Transformer: KonvaNodeComponent<
-  Konva.Transformer,
-  Konva.TransformerConfig
->;
+// export var Circle: KonvaNodeComponent<Konva.Circle, Konva.CircleConfig>;
+// export var Ellipse: KonvaNodeComponent<Konva.Ellipse, Konva.EllipseConfig>;
+// export var Wedge: KonvaNodeComponent<Konva.Wedge, Konva.WedgeConfig>;
+// export var Transformer: KonvaNodeComponent<
+//   Konva.Transformer,
+//   Konva.TransformerConfig
+// >;
 export var Line: KonvaNodeComponent<Konva.Line, Konva.LineConfig>;
-export var Sprite: KonvaNodeComponent<Konva.Sprite, Konva.SpriteConfig>;
-export var Image: KonvaNodeComponent<Konva.Image, Konva.ImageConfig>;
+// export var Sprite: KonvaNodeComponent<Konva.Sprite, Konva.SpriteConfig>;
+// export var Image: KonvaNodeComponent<Konva.Image, Konva.ImageConfig>;
 export var Text: KonvaNodeComponent<Konva.Text, Konva.TextConfig>;
-export var TextPath: KonvaNodeComponent<Konva.TextPath, Konva.TextPathConfig>;
-export var Star: KonvaNodeComponent<Konva.Star, Konva.StarConfig>;
-export var Ring: KonvaNodeComponent<Konva.Ring, Konva.RingConfig>;
-export var Arc: KonvaNodeComponent<Konva.Arc, Konva.ArcConfig>;
-export var Tag: KonvaNodeComponent<Konva.Tag, Konva.TagConfig>;
+// export var TextPath: KonvaNodeComponent<Konva.TextPath, Konva.TextPathConfig>;
+// export var Star: KonvaNodeComponent<Konva.Star, Konva.StarConfig>;
+// export var Ring: KonvaNodeComponent<Konva.Ring, Konva.RingConfig>;
+// export var Arc: KonvaNodeComponent<Konva.Arc, Konva.ArcConfig>;
+// export var Tag: KonvaNodeComponent<Konva.Tag, Konva.TagConfig>;
 export var Path: KonvaNodeComponent<Konva.Path, Konva.PathConfig>;
-export var RegularPolygon: KonvaNodeComponent<
-  Konva.RegularPolygon,
-  Konva.RegularPolygonConfig
->;
-export var Arrow: KonvaNodeComponent<Konva.Arrow, Konva.ArrowConfig>;
+// export var RegularPolygon: KonvaNodeComponent<
+//   Konva.RegularPolygon,
+//   Konva.RegularPolygonConfig
+// >;
+// export var Arrow: KonvaNodeComponent<Konva.Arrow, Konva.ArrowConfig>;
 export var Shape: KonvaNodeComponent<Konva.Shape, Konva.ShapeConfig>;
 
 export var useStrictMode: (useStrictMode: boolean) => void;
diff --git a/es/ReactKonva.js b/es/ReactKonva.js
index dea39471d3e537c29dfd195a2d3044e100a80641..9b24291c4fbc96c63704355525dc41624f130723 100644
--- a/es/ReactKonva.js
+++ b/es/ReactKonva.js
@@ -6,5 +6,25 @@
  * MIT
  */
 'use strict';
-import 'konva';
-export * from './ReactKonvaCore';
+// so we need to import required shapes manually
+import "konva/lib/Core";
+import "konva/lib/shapes/Rect"
+import "konva/lib/shapes/Label"
+import "konva/lib/shapes/Text"
+import "konva/lib/shapes/Path"
+import "konva/lib/shapes/Line"
+
+export {
+  Stage,
+  Layer,
+  FastLayer,
+  Shape,
+  KonvaRenderer,
+// Shapes
+  Label,
+  Rect,
+  Text,
+  Group,
+  Path,
+  Line
+} from "./ReactKonvaCore"
\ No newline at end of file
