diff --git a/package.json b/package.json
index dbab8c9292947a27478455e116188a7f073cb572..e95dd0fceeb164c159db9ade501249a475e577eb 100644
--- a/package.json
+++ b/package.json
@@ -9,15 +9,16 @@
     "test": "mocha",
     "test:exports": "cd test/exports && npm i && npm run test"
   },
-  "main": "./index.js",
-  "module": "./index.mjs",
-  "types": "./index.d.ts",
   "exports": {
     ".": {
       "types": "./index.d.ts",
       "require": "./index.js",
       "import": "./index.mjs"
     },
+    "./patch": {
+      "types": "./patch.d.ts",
+      "import": "./patch.mjs"
+    },
     "./*": {
       "types": "./*.d.ts",
       "require": "./*.js"
diff --git a/patch.d.ts b/patch.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..02661c36fb3283b602a80ae5469c93027876597b
--- /dev/null
+++ b/patch.d.ts
@@ -0,0 +1,13 @@
+export {
+  DefaultMap,
+  DefaultWeakMap,
+  LinkedList,
+  Queue,
+  Stack,
+  MultiMap
+} from './index';
+export {
+  intersection,
+  symmetricDifference,
+  union
+} from './set';
diff --git a/patch.mjs b/patch.mjs
new file mode 100644
index 0000000000000000000000000000000000000000..ca4e93dbba841a307679ae32f2deda0a28717486
--- /dev/null
+++ b/patch.mjs
@@ -0,0 +1,11 @@
+export { default as DefaultMap } from './default-map.js';
+export { default as DefaultWeakMap } from './default-weak-map.js';
+export { default as LinkedList } from './linked-list.js';
+export { default as Queue } from './queue.js';
+export { default as MultiMap } from './multi-map.js';
+export {
+  intersection,
+  symmetricDifference,
+  union
+} from './set.js';
+export { default as Stack } from './stack.js';
