diff --git a/package.json b/package.json
index dbab8c9292947a27478455e116188a7f073cb572..e95dd0fceeb164c159db9ade501249a475e577eb 100644
--- a/package.json
+++ b/package.json
@@ -9,15 +9,16 @@
     "test": "mocha",
     "test:exports": "cd test/exports && npm i && npm run test"
   },
-  "main": "./index.js",
-  "module": "./index.mjs",
-  "types": "./index.d.ts",
   "exports": {
     ".": {
       "types": "./index.d.ts",
       "require": "./index.js",
       "import": "./index.mjs"
     },
+    "./patch": {
+      "types": "./patch.d.ts",
+      "import": "./patch.mjs"
+    },
     "./*": {
       "types": "./*.d.ts",
       "require": "./*.js"
diff --git a/patch.d.ts b/patch.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..d21a135574a27d9572660bfaf03ff24ea2f7b07e
--- /dev/null
+++ b/patch.d.ts
@@ -0,0 +1,5 @@
+export * from './index';
+export {
+  intersection,
+   symmetricDifference, union
+} from './set';
diff --git a/patch.mjs b/patch.mjs
new file mode 100644
index 0000000000000000000000000000000000000000..2a51ca9f39c76a0d47453ba2beb6254c8a38a6fc
--- /dev/null
+++ b/patch.mjs
@@ -0,0 +1,8 @@
+export { default as DefaultMap } from './default-map.js';
+export { default as DefaultWeakMap } from './default-weak-map.js';
+export { default as LinkedList } from './linked-list.js';
+export { default as Queue } from './queue.js';
+export {
+  intersection, symmetricDifference, union
+} from './set.js';
+export { default as Stack } from './stack.js';
